pkg/app/piped/executor/analysis/metrics_analyzer.go

```diff
@@ -114,6 +142,149 @@ func (a *metricsAnalyzer) analyzeWithThreshold(ctx context.Context) (expected bo
 		return false, nil
 	}
 
-	a.logPersister.Successf("[%s] The query result is expected one. Performed query: %q", a.id, a.cfg.Query)
 	return true, nil
 }
+
+// Return false if primary deviates in the specified direction compared to the previous deployment.
+// Return an error if the evaluation could not be executed normally.
+// elapsedTime is used to compare metrics at the same point in time after the analysis has started.
+func (a *metricsAnalyzer) analyzeWithPrevious(ctx context.Context) (bool, error) {
+	now := time.Now()
+	queryRange := metrics.QueryRange{
+		From: now.Add(-a.cfg.Interval.Duration()),
+		To:   now,
+	}
+	points, err := a.provider.QueryPoints(ctx, a.cfg.Query, queryRange)
+	if err != nil {
+		return false, fmt.Errorf("failed to run query: %w", err)
+	}
+
+	prevMetadata, err := a.analysisResultStore.GetLatestAnalysisResult(ctx)
+	if err != nil {
+		return false, fmt.Errorf("failed to fetch the most recent successful analysis metadata: %w", err)
```

Comment:
Should we return ANALYSIS stage with a SUCCESS status when there was no previous deployment?
This could be the first deployment.

Reply:
Right

Reply:
better to separate this error and only return SUCCESS in case `filestore.ErrNotFound` is returned.